virtual_ecosystem/models/animals/animal_model.py

```diff
@@ -239,3 +294,37 @@ def calculate_litter_additions(self) -> dict[str, DataArray]:
                 dims="cell_id",
             ),
         }
+
+    def update_population_densities(self) -> None:
+        """Updates the densities for each functional group in each community."""
+
+        for community_id, community in self.communities.items():
+            for fg_name, cohorts in community.animal_cohorts.items():
+                for cohort in cohorts:  # Now correctly iterating over each cohort
+                    # Calculate the population density for the cohort
+                    population_density = self.calculate_density_for_cohort(cohort)
+
+                    # Update the corresponding entry in the data variable
+                    self.data["population_densities"].loc[
+                        {"community_id": community_id, "functional_group_id": fg_name}
+                    ] = population_density
```

Comment:
There's something missing here. In `cohorts` there might be more than one `cohort`, but they are all going to update the same entry in the data variable, defined by `{"community_id": community_id, "functional_group_id": fg_name}`. Therefore the last cohort processed will overwrite the population density of the previous cohorts.

Reply:
I think I got it sorted out. Let me know if I've gotten confused about something.